### PR TITLE
Change getrows -> subset

### PR DESCRIPTION
**Pull request by quinnj**
Also adds a generic implementation of subset for column-oriented tables
and row-oriented tables that are AbstractVector. Adds implementation and
tests for DictRowTable/DictColumnTable.

### CODE REVIEW

File: src/Tables.jl [hunk 1]
```diff
@@ -587,11 +587,37 @@ or an independent copy:
...
 Support for `view=true` or `view=false` is optional
 (i.e. implementations might error on them if they are not supported).
 """
-function getrows end
+function subset(x::T, inds; view::Union{Bool, Nothing}=nothing) where {T}
+    # because this method is being called, we know `x` didn't define it's own Tables.subset
+    # first check if it supports column access, and if so, apply inds and wrap columns in a DictColumnTable
+    if columnaccess(x)
+        cols = columns(x)
+        if inds isa Integer
```
Comment (bkamins) [inline src/Tables.jl:599]: `Bool` should be excluded here and throw an error. (or maybe `ColumnsRow(cols, inds)` already does it?)

File: src/Tables.jl [hunk 2]
```diff
@@ -587,11 +587,37 @@ or an independent copy:
...
+            return ColumnsRow(cols, inds)
+        else
+            ret = view === true ? _map(c -> Base.view(c, inds), cols) : _map(c -> c[inds], cols)
+            return DictColumnTable(schema(cols), ret)
```
Comment (bkamins) [inline src/Tables.jl:603]: does `DictColumnTable` check that it gets vectors as columns?

Comment (bkamins) [inline src/Tables.jl:603]: As a special case someone can pass `CartesianIndex` as `inds` - do we want to handle it in the `ColumnsRow` case or error on it here?

File: src/dicts.jl [hunk 3]
```diff
@@ -122,6 +124,16 @@ function Base.iterate(x::DictRowTable, st=1)
     return DictRow(x.names, x.values[st]), st + 1
 end
 
+function subset(x::DictRowTable, inds; view::Union{Bool,Nothing} = nothing)
```
Comment (bkamins) [inline src/dicts.jl:127]: similar comments as above to `inds` apply here

File: src/Tables.jl (hunk 1 above)
Comment (quinnj) [reply src/Tables.jl:599]: Ah, great catch.

Comment (quinnj) [reply src/Tables.jl:599]: It is already covered

File: src/Tables.jl (hunk 2 above)
Comment (quinnj) [reply src/Tables.jl:603]: Looks like our existing `vectorcheck` handles that

**Comment by ablaom** [comment]
https://github.com/JuliaData/Tables.jl/pull/284#issuecomment-1238687508

**Pull request merged**